ansible_builder/cli.py

```diff
@@ -100,6 +101,11 @@ def parse_args(args=sys.argv[1:]):
                        default=constants.default_container_runtime,
                        help='Specifies which container runtime to use. Defaults to {0}.'.format(constants.default_container_runtime))
 
+        p.add_argument('--build-arg',
+                       action=BuildArgAction,
+                       default={},
+                       dest='build_args')
```

Comment:
I don't get how this jives with the existing `-b` option. The `-b` is the default, so why would someone ever intend to provide this, but not that?

Reply:
@AlanCoding The main idea here, is we do not want to re-generate a new Dockerfile if we change the base image, otherwise for downstream stuff We'll need 2 git branches for say stable-2.9 / stable-2.11 images of ansible-runner.

https://github.com/ansible/network-ee/pull/11 is an example of how we could use a single branch, but at job level pass in different ansible_runner images via build-args.

Reply:
@AlanCoding `-b` is shorthand for `--base-image`. This new option does not provide a shorthand option (neither do Docker or Podman for this feature).

Reply:
I've also pushed an update to the docs to better explain this.

Reply:
Sorry, I should have been more specific.

I'm not questioning the utility of this feature, I understand its use case.

Assuming this is merged, I'm questioning the utility of the `-b` option. Before you would have done `-b foo_image:latest`, but after the change you are making, that is redundant with using `--build-arg ANSIBLE_RUNNER_IMAGE=foo_image:latest`.

So to make a prescriptive suggestion, lose the `-b` option. Keep its mention you added in the docs, but offer the newer syntax.

Everything about the base image option will cause friction with users. It suggests that we will support other build chains (we won't) or that we would support changing important things in ours like the OS (we won't) and with multi-stage builds there's even more fog about _which_ image it refers to (which the new syntax is more clear about).

I think this project is young enough that we can get away with removing an option. We are changing some pretty fundamental things in other ways, so removing it should be small potatoes.